sdk/armcore/poller.go

```diff
@@ -0,0 +1,1072 @@
+	case http.MethodDelete:
+		pt := &pollingTrackerDelete{pollingTrackerBase: pollingTrackerBase{PollerType: pollerType, FinalStateVia: finalState, OriginalURI: resp.Request.URL.String(), resp: resp, errorHandler: errorHandler}}
+		if err := pt.initializeState(); err != nil {
+			return pt, err
+		}
+		// this initializes the polling header values, we do this during creation in case the
+		// initial response send us invalid values; this way the API call will return a non-nil
+		// error (not doing this means the error shows up in Future.Done)
+		return pt, pt.updatePollingMethod()
+	case http.MethodPatch:
+		pt := &pollingTrackerPatch{pollingTrackerBase: pollingTrackerBase{PollerType: pollerType, FinalStateVia: finalState, OriginalURI: resp.Request.URL.String(), resp: resp, errorHandler: errorHandler}}
+		if err := pt.initializeState(); err != nil {
+			return pt, err
+		}
+		// this initializes the polling header values, we do this during creation in case the
+		// initial response send us invalid values; this way the API call will return a non-nil
+		// error (not doing this means the error shows up in Future.Done)
+		return pt, pt.updatePollingMethod()
+	case http.MethodPost:
+		pt := &pollingTrackerPost{pollingTrackerBase: pollingTrackerBase{PollerType: pollerType, FinalStateVia: finalState, OriginalURI: resp.Request.URL.String(), resp: resp, errorHandler: errorHandler}}
+		if err := pt.initializeState(); err != nil {
+			return pt, err
+		}
+		// this initializes the polling header values, we do this during creation in case the
+		// initial response send us invalid values; this way the API call will return a non-nil
+		// error (not doing this means the error shows up in Future.Done)
+		return pt, pt.updatePollingMethod()
+	case http.MethodPut:
+		pt := &pollingTrackerPut{pollingTrackerBase: pollingTrackerBase{PollerType: pollerType, FinalStateVia: finalState, OriginalURI: resp.Request.URL.String(), resp: resp, errorHandler: errorHandler}}
+		if err := pt.initializeState(); err != nil {
+			return pt, err
+		}
+		// this initializes the polling header values, we do this during creation in case the
+		// initial response send us invalid values; this way the API call will return a non-nil
+		// error (not doing this means the error shows up in Future.Done)
+		return pt, pt.updatePollingMethod()
+	default:
+		return nil, fmt.Errorf("unsupported HTTP method %s", resp.Request.Method)
+	}
+}
+
+// ResumePoller creates a polling tracker from a resume token string.
+// NOTE: this is only meant for internal use in generated code.
+func ResumePoller(pollerType string, token string, errorHandler methodErrorHandler) (Poller, error) {
+	// unmarshal into JSON object to determine the tracker type
+	obj := map[string]interface{}{}
+	err := json.Unmarshal([]byte(token), &obj)
+	if err != nil {
+		return nil, err
+	}
+	if obj["pollerType"] != pollerType {
+		return nil, fmt.Errorf("cannot resume from this poller type")
```

Comment:
Probably good to include the value of `obj["pollerType"]` for extra diagnostics.

Reply:
Done